ompi/mca/common/ompio/common_ompio_file_write.c

```diff
@@ -116,7 +115,7 @@ int mca_common_ompio_file_write (ompio_file_t *fh,
     else {
 	bytes_per_cycle = OMPIO_MCA_GET(fh, cycle_buffer_size);
     }
-    cycles = ceil((float)max_data/bytes_per_cycle);
+    cycles = (max_data + bytes_per_cycle - 1) / bytes_per_cycle;
```

Comment:
From reviewing the types, `cycles` here should, as in `read`, be at least `size_t` instead of `int`.

Saver would be to make all byte counters in both methods fixed to `uint64_t`...

Anyway, `int`'s range is on 64bit architectures/data models smaller than the range of `size_t` and at least read/write should implement the same overflow ranges :)